ckanext/spatial/tests/test_harvest.py

```diff
@@ -0,0 +1,1165 @@
+
+        # Package was not updated
+        assert second_package_dict, first_package_dict['id'] == second_package_dict['id']
+        assert not second_obj.package, not second_obj.package_id
+        assert second_obj.current == False, first_obj.current == True
+
+        # Create and run a third job, forcing the importing to simulate an update in the package
+        third_job = self._create_job(source.id)
+        third_obj = self._run_job_for_single_document(third_job,force_import=True)
+
+        # For some reason first_obj does not get updated after the import_stage,
+        # and we have to force a refresh to get the actual DB values.
+        Session.remove()
+        Session.add(first_obj)
+        Session.add(second_obj)
+        Session.add(third_obj)
+
+        Session.refresh(first_obj)
+        Session.refresh(second_obj)
+        Session.refresh(third_obj)
+
+        third_package_dict = get_action('package_show')(self.context,{'id':third_obj.package_id})
+
+        # Package was updated
+        assert third_package_dict, first_package_dict['id'] == third_package_dict['id']
+        assert third_obj.package, third_obj.package_id == first_package_dict['id']
+        assert third_obj.current == True
+        assert second_obj.current == False
+        assert first_obj.current == False
+
+    def test_harvest_deleted_record(self):
+
+        # Create source
+        source_fixture = {
+            'title': 'Test Source',
+            'name': 'test-source',
+            'url': u'http://127.0.0.1:8999/gemini2.1/service1.xml',
+            'source_type': u'gemini-single'
+        }
+
+        source, first_job = self._create_source_and_job(source_fixture)
+
+        first_obj = self._run_job_for_single_document(first_job)
+
+        first_package_dict = get_action('package_show')(self.context,{'id':first_obj.package_id})
+
+        # Package was created
+        assert first_package_dict
+        assert first_package_dict['state'] == u'active'
+        assert first_obj.current == True
+
+        # Delete package
+        first_package_dict['state'] = u'deleted'
+        self.context.update({'id':first_package_dict['id']})
+        updated_package_dict = get_action('package_update')(self.context,first_package_dict)
+
+        # Create and run a second job, the date has not changed, so the package should not be updated
+        # and remain deleted
+        first_job.status = u'Finished'
+        first_job.save()
+        second_job = self._create_job(source.id)
+
+        second_obj = self._run_job_for_single_document(second_job)
+
+        second_package_dict = get_action('package_show')(self.context,{'id':first_obj.package_id})
+
+        # Package was not updated
+        assert second_package_dict, updated_package_dict['id'] == second_package_dict['id']
+        assert not second_obj.package, not second_obj.package_id
+        assert second_obj.current == False, first_obj.current == True
```

Comment:
```suggestion
        assert second_obj.current == False
        assert first_obj.current == True
```
This does not do what you think it does. These two asserts need to be separate, the part after the comma is the error raised if the assertion fails, usually a string describing the error.

Reply:
same for all of these asserts, actually